### PR TITLE
Refactor - add sqd providers

### PR DESCRIPTION
**Pull request by RutPuselj**

### CODE REVIEW

File: dist/AbstractApi.js [hunk 1]
```diff
@@ -8,7 +8,8 @@ const AuthService_1 = __importDefault(require("./AuthService"));
 class AbstractApi {
     constructor(apiRoot, requiresAuth, initOptions = {}) {
         this.hasAuthService = false;
-        const { getToken, authScheme, loadingProvider } = initOptions;
+        const { getToken, authScheme } = initOptions;
+        this.sqdProvider = [];
```
Comment (IvanAxilis) [inline dist/AbstractApi.js]: sqdProviders :)

File: dist/AbstractApi.js [hunk 2]
```diff
@@ -83,7 +72,26 @@ class AbstractApi {
...
+        return this;
+    }
+    hasSqdProvider(provider) {
+        return this.sqdProvider.includes(provider);
+    }
+    indexOfSqdProvider(provider) {
+        return this.sqdProvider.indexOf(provider);
+    }
+    deleteSqdProvider(provider) {
+        const index = this.indexOfSqdProvider(provider);
```
Comment (IvanAxilis) [inline dist/AbstractApi.js:88]: what if there is no such provider

**Review by IvanAxilis — CHANGES_REQUESTED** [review]

**Review by IvanAxilis — APPROVED** [review]

File: src/AbstractApi.ts [hunk 3]
```diff
@@ -109,12 +96,32 @@ export default abstract class AbstractApi {
     const headers = options && options.headers ? options.headers : {};
     return options ? { ...options, method, headers, body: options.body } : { method, headers };
   }
-}
 
-interface IRequestOptions extends RequestInit {
-  handleLoading?: boolean;
-}
+  public addSqdProvider(provider:ISqdProvider): AbstractApi {
```
Comment (marinaRupe) [inline src/AbstractApi.ts]: Add a space before `ISqdProvider` (multiple places)

File: src/AbstractApi.ts [hunk 4]
```diff
@@ -109,12 +96,32 @@ export default abstract class AbstractApi {
...
+
+  public deleteSqdProvider(provider:ISqdProvider): boolean {
+    const index = this.indexOfSqdProvider(provider);
+    if (index < 0) return false;
+    return this.deleteSqdProviderByIndex(index);
+  }
 
-class FetchError extends Error {
-  response?: any;
+  public deleteSqdProviderByIndex(index:number): boolean {
```
Comment (marinaRupe) [inline src/AbstractApi.ts]: Add a space before ` number`

**Review by marinaRupe — CHANGES_REQUESTED** [review]

**Review by marinaRupe — APPROVED** [review]

**Pull request merged**